source/addonHandler/__init__.py

```diff
@@ -144,11 +160,10 @@ def load(self) -> None:
 			# For the installer, this is not written to disk.
 			# Portable/temporary copies will write this on the first run.
 			# Mark overridden compatible add-ons as blocked.
-			state[AddonStateCategory.BLOCKED].update(state[AddonStateCategory.OVERRIDE_COMPATIBILITY])
+			self[AddonStateCategory.BLOCKED].update(self[AddonStateCategory.OVERRIDE_COMPATIBILITY])
 			# Reset overridden compatibility for add-ons that were overridden by older versions of NVDA.
-			state[AddonStateCategory.OVERRIDE_COMPATIBILITY].clear()
+			self[AddonStateCategory.OVERRIDE_COMPATIBILITY].clear()
 		self.manualOverridesAPIVersion = MajorMinorPatch(*addonAPIVersion.BACK_COMPAT_TO)
-		self.update(state)

```

Comment:
is there a reason why using `state` instead of `self` was changed and calling `update` was removed?
The current design was intentional and I believe this introduces and unnecessary risk to the beta. We have limited time to test upgrading and downgrading NVDA with add-ons as per work in #15439

Reply:
There is simply no `state` local variable anymore. Now the data  is loaded into the instance by `fromPickledDict`, which means that the loading from the dictionary can be tested independently in unit tests. The process is covered by unit tests, which is certainly an improvement to what we had previously. Reading them should give you confidence that functionally the process remained the same.